src/main/java/jenkins/plugins/openstack/compute/JCloudsRetentionStrategy.java

```diff
@@ -19,7 +23,7 @@
 /**
  * @author Vijay Kiran
  */
-public class JCloudsRetentionStrategy extends RetentionStrategy<JCloudsComputer> {
+public class JCloudsRetentionStrategy extends RetentionStrategy<JCloudsComputer> implements ExecutorListener {
```

Comment:
Seems like a good idea; but I'd like to use `node#canTake` *in addition* to what I'm doing here, not instead of it.  My reasoning is that the current code has the nice property of disposing of "burned" slaves instantly upon task completion (i.e. no need to wait until the cron task removes it). But I definitely like your idea to use `node#canTake` to ensure that there are no race conditions: a flag controlling the behaviour of `node#canTake` would be set on the slave at the *start* of the first task, thus guaranteeing that no task will ever run on the slave.
What do you think?

Reply:
Exactly